### PR TITLE
(PUP-8314) add decorations to raise ArgumentError strings part 5

### PR DESCRIPTION
**Pull request by ericldelaney**

### CODE REVIEW

File: lib/puppet/interface/option.rb [hunk 1]
```diff
@@ -33,20 +33,28 @@ def initialize(parent, *declaration, &block)
...
         end
         if dup = dups[name] then
-          raise ArgumentError, "#{item.inspect}: duplicates existing alias #{dup.inspect} in #{@parent}"
+          # TODO
+          # TODO
+          # TODO
+          # TODO
+          # TODO
+          # TODO
+          # TODO
```
Comment (melissa) [inline lib/puppet/interface/option.rb]: This is a lot of todos....

Comment (ericldelaney) [reply lib/puppet/interface/option.rb]: Sorry that was a marker for me to know where to start looking again and I missing pushing up the delete for it.

File: lib/puppet/interface/action_builder.rb [hunk 2]
```diff
@@ -144,6 +150,9 @@ def initialize(face, name, &block)
     @face   = face
     @action = Puppet::Interface::Action.new(face, name)
     instance_eval(&block)
-    @action.when_invoked or raise ArgumentError, "actions need to know what to do when_invoked; please add the block"
+    unless @action.when_invoked
+      #TRANSLATORS 'when_invoked' is a method name and should not be translated and 'block' is a Ruby code block
```
Comment (melissa) [inline lib/puppet/interface/action_builder.rb]: Is the comment here just adding context to help translate `block`, or should `block` not be translated?

Comment (melissa) [reply lib/puppet/interface/action_builder.rb]: Also, I don't like this error message, I think it doesn't lend itself well to translation. But that's probably neither here nor there. Definitely not something I think we should fix here.

File: lib/puppet/interface/documentation.rb [hunk 3]
```diff
@@ -56,7 +56,8 @@ module TinyDocs
     # @dsl Faces
     attr_doc :summary do |value|
       value =~ /\n/ and
-        raise ArgumentError, "Face summary should be a single line; put the long text in 'description' instead."
+        #TRANSLATORS 'Face' refers to a programming API in Puppet
```
Comment (melissa) [inline lib/puppet/interface/documentation.rb]: I feel like you should add `and should not be translated` here. Also, `summary` and `description` are key values that we grab data from. I have no idea for sure, but I have a feeling those are going to need to be in English as well

File: lib/puppet/interface/documentation.rb [hunk 4]
```diff
@@ -200,11 +201,11 @@ def short_description(value = nil)
     def author(value = nil)
       unless value.nil? then
         unless value.is_a? String
-          raise ArgumentError, 'author must be a string; use multiple statements for multiple authors'
+          raise ArgumentError, _('author must be a string; use multiple statements for multiple authors')
```
Comment (melissa) [inline lib/puppet/interface/documentation.rb]: Same as above for `author`, though again this is definitely debatable. I'd be fine merging this in as is

File: lib/puppet/interface/documentation.rb [hunk 5]
```diff
@@ -243,7 +244,7 @@ def author=(value)
     # @dsl Faces
     def copyright(owner = nil, years = nil)
       if years.nil? and not owner.nil? then
-        raise ArgumentError, 'copyright takes the owners names, then the years covered'
+        raise ArgumentError, _('copyright takes the owners names, then the years covered')
```
Comment (melissa) [inline lib/puppet/interface/documentation.rb]: Same again for `copyright`

File: lib/puppet/interface/documentation.rb (hunk 3 above)
Comment (ericldelaney) [reply lib/puppet/interface/documentation.rb]: Ok, I agree

File: lib/puppet/interface/documentation.rb (hunk 5 above)
Comment (ericldelaney) [reply lib/puppet/interface/documentation.rb]: ok changed

File: lib/puppet/interface/documentation.rb (hunk 4 above)
Comment (ericldelaney) [reply lib/puppet/interface/documentation.rb]: yeah, changed

File: lib/puppet/interface/documentation.rb (hunk 3 above)
Comment (jtappa) [reply lib/puppet/interface/documentation.rb]: Too bad we don't have a text decoration for words that won't need translation, would help with so many duplicated translator notes

File: lib/puppet/interface/action_builder.rb (hunk 2 above)
Comment (ericldelaney) [reply lib/puppet/interface/action_builder.rb]: I'm inclined to leave this as is.

**Comment by ericldelaney** [comment]
@melissa Updated

File: lib/puppet/pops/model/factory.rb [hunk 6]
```diff
@@ -790,7 +790,9 @@ def self.NUMBER(name_or_numeric)
       end
     else
       # Bad number should already have been caught by lexer - this should never happen
-      raise ArgumentError, "Internal Error, NUMBER token does not contain a valid number, #{name_or_numeric}"
+      #TRANSLATORS 'NUMBER' refers to a method name and the 'name_or_numeric' was the passed in value
```
Comment (jtappa) [inline lib/puppet/pops/model/factory.rb]: should `NUMBER` and `name_or_numberic` also not be translated, and should that be noted in this comment?

File: lib/puppet/pops/serialization/abstract_writer.rb [hunk 7]
```diff
@@ -99,7 +99,8 @@ def write_tpl_qname(ep, qname)
   #
   # @api private
   def write_tpl(ep, value)
-    raise ArgumentError, 'Internal error. Integers cannot be tabulated in extension payload' if value.is_a?(Integer)
+    #TRANSLATORS 'Integers' is a Ruby class for numbers
```
Comment (jtappa) [inline lib/puppet/pops/serialization/abstract_writer.rb]: Should this one also say, "and should not be translated" ?

File: lib/puppet/pops/loader/module_loaders.rb [hunk 8]
```diff
@@ -98,7 +98,8 @@ def initialize(parent_loader, loaders, module_name, path, loader_name, loadables
       @loaders = loaders
       @loadables = loadables
       unless (loadables - LOADABLE_KINDS).empty?
-        raise ArgumentError, 'given loadables are not of supported loadable kind'
+        #TRANSLATORS 'loadables' is a variable containing loadable modules
```
Comment (jtappa) [inline lib/puppet/pops/loader/module_loaders.rb]: Should this one also say, "and should not be translated" ? Looks like `loadables` is also a variable above

Comment (ericldelaney) [reply lib/puppet/pops/loader/module_loaders.rb]: done

File: lib/puppet/pops/model/factory.rb (hunk 6 above)
Comment (ericldelaney) [reply lib/puppet/pops/model/factory.rb]: done

File: lib/puppet/pops/serialization/abstract_writer.rb (hunk 7 above)
Comment (ericldelaney) [reply lib/puppet/pops/serialization/abstract_writer.rb]: I waffled on this one, since you brought it up. I'll add it.

**Comment by ericldelaney** [comment]
@jtappa Updated

File: lib/puppet/pops/serialization/abstract_writer.rb (hunk 7 above)
Comment (jtappa) [reply lib/puppet/pops/serialization/abstract_writer.rb]: It sucks that it has to be said in so many comments and there isn't a better way to notate it, but might as well be consistent I guess ¯\_(ツ)_/¯

File: lib/puppet/interface/action_builder.rb (hunk 2 above)
Comment (melissa) [reply lib/puppet/interface/action_builder.rb]: 👍

**Pull request merged**